### PR TITLE
Ren branch

### PR DESCRIPTION
**Pull request by chuckrenny**

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -23,6 +23,40 @@ def initialize(files)
...
+  # 1. pull out all game_id for a season in @games as an array
+  # 2. iterate over @game_teams using our created array of game_id as reference
+    # make hash of team_id as keys and set values to [goals + itself, shots + itself]
+  # 3. use transform_values on hash to give goals / shots .round(4)
+  # 4. Find the max value and subsequent key and return the team_name with team_list
+
+  # 1. array of all game_id from a season
+  all_season_game_id = @games.map do |game|
+    game.game_id if game.season == season
+  end.compact
```
Comment (Antoine-Aube) [inline lib/stat_tracker.rb]: Did you have to use compact because some values returned nil? What values were those?

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -23,6 +23,40 @@ def initialize(files)
...
+  def most_accurate_team(season)
+  # name of team with the best ratio of shots to goals for the season 
+  # 1. pull out all game_id for a season in @games as an array
+  # 2. iterate over @game_teams using our created array of game_id as reference
+    # make hash of team_id as keys and set values to [goals + itself, shots + itself]
+  # 3. use transform_values on hash to give goals / shots .round(4)
+  # 4. Find the max value and subsequent key and return the team_name with team_list
+
+  # 1. array of all game_id from a season
+  all_season_game_id = @games.map do |game|
```
Comment (Antoine-Aube) [inline lib/stat_tracker.rb]: probs can bring out as helper in refactor

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -23,6 +23,40 @@ def initialize(files)
...
+    game.game_id if game.season == season
+  end.compact
+
+  # 2. hash of team_id as keys and [goals, values] as values
```
Comment (Antoine-Aube) [inline lib/stat_tracker.rb]: values represents game.shots here yeah?

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -23,6 +23,40 @@ def initialize(files)
...
+  team_id_goals_shots = @game_teams.each_with_object(Hash.new([0,0])) do |game, hash|
+    # if game.game_id exists in our all_season_game_id array, then create key-value pair
+    if all_season_game_id.include?(game.game_id)
+      hash[game.team_id] = [game.goals + hash[game.team_id][0], game.shots + hash[game.team_id][1]]
```
Comment (Antoine-Aube) [inline lib/stat_tracker.rb]: incrementing the goals and shots? This is a bit brain bendy but I think I understand what's happening here. Might have to chat this one over for further understanding, but seems to be working!

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -23,6 +23,40 @@ def initialize(files)
...
+  end
+  
+  # 3. Calculate the average of goals/shots and assign to values
+  avg_goals_made = team_id_goals_shots.transform_values do |value|
+    (value[0] / value[1].to_f).round(4)
+  end
+
+  # 4. Find max value and subsequent key to return the team_name with helper method team_list
+  team_name = avg_goals_made.key(avg_goals_made.values.max)
+  team_list[team_name]
```
Comment (Antoine-Aube) [inline lib/stat_tracker.rb]: nice I reusedd team list as well.

File: lib/stat_tracker.rb (hunk 1 above)
Comment (chuckrenny) [reply lib/stat_tracker.rb]: Compact is needed to remove all nil values that result from not passing the conditional logic on line 36.

**Pull request merged**